### PR TITLE
IPSEC backend

### PR DESCRIPTION
**Pull request by eyakubovich**
Updated version of https://github.com/coreos/flannel/pull/290

This is still a WIP on the packaging front. `charon` daemon depends on `libatomic` wich is missing from the base image.

### CODE REVIEW

**Comment by tomdee** [comment]
@eyakubovich Congratulations on the new job! Are you looking for someone to take over this PR now?

**Comment by eyakubovich** [comment]
@tomdee Thanks! I was temporarily helping out CoreOS and took some time to update the PR. However I am no longer working on this. On a personal level, I would love it if this PR finally got merged, @MohdAhmad worked very hard on it and I think it would be a good addition to flannel. Lots of people have asked for this feature.

File: backend/ipsec/handle_charon.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	}
+	remoteAuthConf := goStrongswanVici.AuthConf{
+		AuthMethod: "psk",
+	}
+
+	ikeConf := goStrongswanVici.IKEConf{
+		LocalAddrs:  []string{localLease.Attrs.PublicIP.String()},
+		RemoteAddrs: []string{remoteLease.Attrs.PublicIP.String()},
+		Proposals:   []string{"aes256-sha256-modp4096"},
+		Version:     "1",
```
Comment (matthazinski) [inline backend/ipsec/handle_charon.go]: Is there an advantage to using IKEv1 here? IKEv2 seems to be simpler and more reliable.

**Comment by aquavitae** [comment]
Any progress on this?

File: backend/ipsec/handle_charon.go (hunk 1 above)
Comment (mkutsevol) [inline backend/ipsec/handle_charon.go]: Looks like this was taken directly from library docs. 
https://github.com/bronze1man/goStrongswanVici 
What is wrong with v1 is it just works? 
I'm still running some ikev1 without any problems.

**Comment by mkutsevol** [comment]
I've made a follow-up PR #637

**Comment by tomdee** [comment]
Closing in favour of #637

**Pull request closed without merging**